Gifski/util.swift

```diff
@@ -2515,3 +2515,18 @@ extension NSObject {
 		return type(of: self).simpleClassName
 	}
 }
+
+extension AVPlayerItem {
+	/// The playable range of the item.
+	/// Can be `nil` when the `.duration` is not available, for example, when the asset has not yet been fully loaded or if it's a live stream.
+	var playbackRange: ClosedRange<Double>? {
+		guard duration.isNumeric else {
+			return nil
+		}
+
+		let startTime = reversePlaybackEndTime.isNumeric ? reversePlaybackEndTime.seconds : 0
+		let endTime = forwardPlaybackEndTime.isNumeric ? forwardPlaybackEndTime.seconds : duration.seconds
+
+		return startTime < endTime ? startTime...endTime : endTime...startTime
```

Comment:
```suggestion
		return startTime < endTime ? startTime...endTime : nil
```

this `endTime...startTime` shouldn't probably happen but in case it does, maybe we should just bail and return `nil`?

Reply:
I was making this generic as there are other use-cases than ours where it could happen. For example, when someone sets `reversePlaybackEndTime` manually. I don't really see any downside in *just handling it*. Do you?

Reply:
ah ye, I was just scared of this use-case maybe 😄 let’s try that 👍🏻